core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -149,28 +149,49 @@ protected void validate() {
       (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
       throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
     }
+    if (format.equalsIgnoreCase("text") && schema == null) {
```

Comment:
there's no need to check the format here right? If schema is null then nothing below here needs to be checked